tests/test_dse.py

```diff
@@ -1171,7 +1171,65 @@ def test_extraction_from_lifted_ispace(self):
 
         # Also check against expected operation count to make sure
         # all redundancies have been detected correctly
-        assert summary[('section0', None)].ops == 115
+        assert summary[('section0', None)].ops == 120
+
+    @patch("devito.passes.clusters.aliases.MIN_COST_ALIAS", 1)
+    @switchconfig(profiling='advanced')
+    def test_tti_adjoint_akin(self):
+        """
+        Extrapolated from TTI adjoint.
+        """
+        so = 4
+        to = 2
+        soh = so // 2
+        T = transpose
+
+        grid = Grid(shape=(10, 10, 10), dtype=np.float64)
+        x, y, z = grid.dimensions
+
+        p = TimeFunction(name='p', grid=grid, space_order=so, time_order=to)
+        r = TimeFunction(name='r', grid=grid, space_order=so, time_order=to)
+        delta = Function(name='delta', grid=grid, space_order=so)
+        theta = Function(name='theta', grid=grid, space_order=so)
+        phi = Function(name='phi', grid=grid, space_order=so)
+
+        p.data_with_halo[:] = 1.
+        r.data_with_halo[:] = 0.5
+        delta.data_with_halo[:] = 0.2
+        theta.data_with_halo[:] = 0.8
+        phi.data_with_halo[:] = 0.2
+
+        costheta = cos(theta)
+        sintheta = sin(theta)
+        cosphi = cos(phi)
+        sinphi = sin(phi)
+
+        delta = sqrt(delta)
+
+        field = delta*p + r
+        Gz = -(sintheta * cosphi*first_derivative(field, dim=x, fd_order=soh) +
+               sintheta * sinphi*first_derivative(field, dim=y, fd_order=soh) +
+               costheta * first_derivative(field, dim=z, fd_order=soh))
+        Gzz = (first_derivative(Gz * sintheta * cosphi, dim=x, fd_order=soh, matvec=T) +
+               first_derivative(Gz * sintheta * sinphi, dim=y, fd_order=soh, matvec=T) +
+               first_derivative(Gz * costheta, dim=z, fd_order=soh, matvec=T))
+
+        # Equation
+        eqn = [Eq(r.backward, Gzz)]
+
+        op0 = Operator(eqn, subs=grid.spacing_map, opt=('noop', {'openmp': True}))
+        op1 = Operator(eqn, subs=grid.spacing_map, opt=('advanced', {'openmp': True}))
+
+        # Check numerical output
+        op0(time_M=1)
+        exp_norm_r = norm(r)
+        r.data_with_halo[:] = 0.5
```

Comment:
so `r` is unchanged after op()?

Reply:
what do you mean? `op` computes `r`, that's why I'm re-initializing it to its original value, namely 0.5

Reply:
Oh right ok sorry missread that. Shouldn't it run for mre than just one time step to make sure stuff don't blow up?

Reply:
Well I'm sure this MFE is unstable so if we run for more it might indeed blow up. I'm totally confident that running for two timesteps here is more than enough and fits our needs. If something were screwed at this point, it would be screwed in such a way that you'd have a spike in the computed values (due to reading from OOB or un-initialized data)

Reply:
Sounds abut right 👍